### PR TITLE
Request result

### PR DESCRIPTION
**Pull request by hbjorgo**
Returning Result object from commands and queries. Using controller extension method to return http result.

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Command/ModeCommands/DeleteMode/DeleteModeCommandHandler.cs [hunk 1]
```diff
@@ -17,12 +18,12 @@ public DeleteModeCommandHandler(IModeRepository modeRepository, IUnitOfWork unit
...
         }
 
-        public async Task<Unit> Handle(DeleteModeCommand request, CancellationToken cancellationToken)
+        public async Task<Result<Unit>> Handle(DeleteModeCommand request, CancellationToken cancellationToken)
         {
             var mode = await _modeRepository.GetByIdAsync(request.ModeId);
             _modeRepository.Remove(mode);
             await _unitOfWork.SaveChangesAsync(cancellationToken);
-            return Unit.Value;
+            return new SuccessResult<Unit>(Unit.Value);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/ModeCommands/DeleteMode/DeleteModeCommandHandler.cs]: What is Unit.Value here?

File: src/Equinor.Procosys.Preservation.Query/RequirementTypeAggregate/GetRequirementTypeByIdQueryHandler.cs [hunk 2]
```diff
@@ -3,23 +3,24 @@
...
 
         public GetRequirementTypeByIdQueryHandler(IRequirementTypeRepository requirementTypeRepository) => _requirementTypeRepository = requirementTypeRepository;
 
-        public async Task<RequirementTypeDto> Handle(GetRequirementTypeByIdQuery request, CancellationToken cancellationToken)
+        public async Task<Result<RequirementTypeDto>> Handle(GetRequirementTypeByIdQuery request, CancellationToken cancellationToken)
         {
             var rt = await _requirementTypeRepository.GetByIdAsync(request.Id);
             if (rt == null)
             {
                 throw new NotFoundException($"RequirementType with id {request.Id} not found");
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/RequirementTypeAggregate/GetRequirementTypeByIdQueryHandler.cs]: return new NotFoundResult

File: src/Equinor.Procosys.Preservation.Command/ModeCommands/DeleteMode/DeleteModeCommandHandler.cs (hunk 1 above)
Comment (hbjorgo) [reply src/Equinor.Procosys.Preservation.Command/ModeCommands/DeleteMode/DeleteModeCommandHandler.cs]: That's the Mediatr type for NoContent.

File: src/Equinor.Procosys.Preservation.Query/RequirementTypeAggregate/GetRequirementTypeByIdQueryHandler.cs (hunk 2 above)
Comment (hbjorgo) [reply src/Equinor.Procosys.Preservation.Query/RequirementTypeAggregate/GetRequirementTypeByIdQueryHandler.cs]: Done

**Pull request merged**